# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86-cvs/devfs-vc.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- xfree86-cvs-2003-01-08/programs/Xserver/hw/kdrive/linux/linux.c~	2002-10-08 23:27:55.000000000 +0200
+++ xfree86-cvs-2003-01-08/programs/Xserver/hw/kdrive/linux/linux.c	2003-01-11 21:26:19.000000000 +0100
@@ -79,10 +79,10 @@
 	vtno = kdVirtualTerminal;
     else
     {
-	if ((fd = open("/dev/tty0",O_WRONLY,0)) < 0) 
+	if ((fd = open("/dev/vc/0",O_WRONLY,0)) < 0) 
 	{
 	    FatalError(
-		       "LinuxInit: Cannot open /dev/tty0 (%s)\n",
+		       "LinuxInit: Cannot open /dev/vc/0 (%s)\n",
 		       strerror(errno));
 	}
 	if ((ioctl(fd, VT_OPENQRY, &vtno) < 0) ||
@@ -93,7 +93,7 @@
     }
     close(fd);
 
-    sprintf(vtname,"/dev/tty%d",vtno); /* /dev/tty1-64 */
+    sprintf(vtname,"/dev/vc/%d",vtno); /* /dev/vc/1-64 */
 
     if ((LinuxConsoleFd = open(vtname, O_RDWR|O_NDELAY, 0)) < 0)
     {
@@ -110,7 +110,7 @@
      *
      * Why is this needed?
      */
-    LinuxCheckChown ("/dev/tty0");
+    LinuxCheckChown ("/dev/vc/0");
     /*
      * Linux doesn't switch to an active vt after the last close of a vt,
      * so we do this ourselves by remembering which is active now.
@@ -441,7 +441,7 @@
 	activeVT = -1;
     }
     close(LinuxConsoleFd);                /* make the vt-manager happy */
-    fd = open ("/dev/tty0", O_RDWR|O_NDELAY, 0);
+    fd = open ("/dev/vc/0", O_RDWR|O_NDELAY, 0);
     if (fd >= 0)
     {
 	ioctl (fd, VT_GETSTATE, &vts);
